src/mds/CDir.cc

```diff
@@ -2695,6 +2695,17 @@ bool CDir::contains(CDir *x)
   }
 }
 
+bool CDir::can_rep() const
+{
+  if (!is_rep()) 
+    return true;
+
+  unsigned mds_num = mdcache->mds->get_mds_map()->get_num_mds(MDSMap::STATE_ACTIVE);
+  if ((mds_num - 1) != get_replicas().size()) 
```

Comment:
What about the edge case where the user decides to change max_mds intentionally. Then, won't this condition be true every-time that directory is hit and lead to repeated broadcasts for replication? @ukernel  and @batrick Please do look into this too.